server/src/main/java/io/crate/analyze/relations/FullQualifiedNameFieldProvider.java

```diff
@@ -97,9 +100,23 @@ public Symbol resolveField(QualifiedName qualifiedName, @Nullable List<String> p
             tableNameMatched = true;
 
             AnalyzedRelation sourceRelation = entry.getValue();
-            Symbol newField = sourceRelation.getField(columnIdent, operation);
+            Symbol newField;
+            try {
+                newField = sourceRelation.getField(columnIdent, operation, errorOnUnknownObjectKey);
+            } catch (UnknownObjectKeyExceptionalControlFlow e) {
```

Comment:
I commented on this but somehow Github must have swallowed it :(
Trying again...


That we use the exception for control flow here still bugs me a bit.

Some options I can think of:

- Change the return type of `DocTableInfo.getDynamic` to `Symbol` or something like `Either<DynamicReference, Literal>`.
- Introduce some kind of `VoidReference` that extends `DynamicReference`. I assume the reason we return a Literal.NULL is to optimize the scenario and avoid actually looking up a column we know doesn't exist.

I think I'd prefer the latter. It would probably also solve the `AliasSymbol` workaround requirement